valgrind errors on tcp tests when ucx is configured with --enable-debug

### Describe the bug
Several tcp gtests are failing on valgrind if ucx is compiled with --enable-debug (and configure-devel).
without this flag, they pass.

### Steps to Reproduce
- Command line for example:
```
GTEST_FILTER=tcp/uct_p2p_rma_test.put_zcopy/0 make -C ./test/gtest test_valgrind
```
- UCX master at 13624a0436093c76066a8d4916f03c6480fd60e5

```
[     INFO ] checking dcx: yes
[     INFO ] checking ud: yes
[     INFO ] checking udx: yes
[     INFO ] checking rc: yes
[     INFO ] checking rcx: yes
[     INFO ] checking shm_ib: yes
[     INFO ] checking ugni: no
[     INFO ] checking self: yes
[     INFO ] checking tcp: yes
[     INFO ] checking all: yes
[     INFO ] checking all_rcdc: yes
[     INFO ] checking self_rcx: yes
[     INFO ] checking rc_v: yes
[     INFO ] checking rc_x: yes
[     INFO ] checking ud_mlx5: yes
[     INFO ] checking shm: yes
[     INFO ] checking rc_ud: yes
[     INFO ] checking dc_ud: yes
[     INFO ] checking ib: yes
[     INFO ] checking rc_dc: yes
[     INFO ] checking shm_rc: yes
[     INFO ] checking rcv: yes
==134033== brk segment overflow in thread #1: can't grow to 0x4a2e000
==134033== (see section Limitations in user manual)
[     INFO ] checking no_ud_ud_x: yes
[     INFO ] Using random seed of 1545
[     INFO ]  Setting for valgrind: UCX_IB_RX_QUEUE_LEN=512
[     INFO ]  Setting for valgrind: UCX_IB_RX_BUFS_GROW=512
[     INFO ]  Setting for valgrind: UCX_MM_RX_BUFS_GROW=128
[     INFO ]  Setting for valgrind: UCX_IB_TX_QUEUE_LEN=128
[     INFO ]  Setting for valgrind: UCX_IB_TX_BUFS_GROW=64
[     INFO ]  Setting for valgrind: UCX_RC_TX_CQ_LEN=256
[     INFO ]  Setting for valgrind: UCX_CM_TIMEOUT=600ms
[     INFO ]  Setting for valgrind: UCX_TCP_TX_BUFS_GROW=512
[     INFO ]  Setting for valgrind: UCX_TCP_RX_BUFS_GROW=512
[     INFO ]  Setting for valgrind: UCX_TCP_RX_SEG_SIZE=16k
Note: Google Test filter = tcp/uct_p2p_rma_test.put_zcopy/0
[==========] Running 1 test from 1 test case.
[----------] Global test environment set-up.
[----------] 1 test from tcp/uct_p2p_rma_test
[ RUN      ] tcp/uct_p2p_rma_test.put_zcopy/0 <tcp/eno1>
[     INFO ] memory_type:host 0 20971520 ==134033== WARNING: valgrind ignores shmget(shmflg) SHM_HUGETLB
==134033== Source and destination overlap in memcpy(0xba3d000, 0xba3d005, 20)
==134033==    at 0x4C2E00C: memcpy@@GLIBC_2.14 (vg_replace_strmem.c:1018)
==134033==    by 0x53CFDF6: uct_tcp_ep_handle_put_req (tcp_ep.c:834)
==134033==    by 0x53D0212: uct_tcp_ep_progress_am_rx (tcp_ep.c:906)
==134033==    by 0x53D2948: uct_tcp_ep_progress_rx (tcp.h:490)
==134033==    by 0x53D2DB9: uct_tcp_iface_handle_events (tcp_iface.c:194)
==134033==    by 0x509C14E: ucs_event_set_wait (event_set.c:213)
==134033==    by 0x53D2E5F: uct_tcp_iface_progress (tcp_iface.c:211)
==134033==    by 0x61A4F3: ucs_callbackq_dispatch (callbackq.h:211)
==134033==    by 0x62348F: uct_worker_progress (uct.h:2204)
==134033==    by 0x62348F: uct_test::entity::progress() const (uct_test.cc:838)
==134033==    by 0x61FF9B: uct_test::progress() const (uct_test.cc:581)
==134033==    by 0x618AA5: uct_p2p_test::blocking_send(ucs_status_t (uct_p2p_test::*)(uct_ep*, uct_test::mapped_buffer const&, uct_test::mapped_buffer const&), uct_ep*, uct_test::mapped_buffer const&, uct_test::mapped_buffer const&, bool) (uct_p2p_test.cc:277)
==134033==    by 0x5F9686: uct_p2p_rma_test::test_xfer(ucs_status_t (uct_p2p_test::*)(uct_ep*, uct_test::mapped_buffer const&, uct_test::mapped_buffer const&), unsigned long, unsigned int, ucs_memory_type) (test_p2p_rma.cc:81)
==134033==    by 0x6192ED: void uct_p2p_test::test_xfer_print<ucs::detail::message_stream>(ucs::detail::message_stream&, ucs_status_t (uct_p2p_test::*)(uct_ep*, uct_test::mapped_buffer const&, uct_test::mapped_buffer const&), unsigned long, unsigned int, ucs_memory_type) (uct_p2p_test.cc:137)
==134033==    by 0x6183B2: uct_p2p_test::test_xfer_multi_mem_type(ucs_status_t (uct_p2p_test::*)(uct_ep*, uct_test::mapped_buffer const&, uct_test::mapped_buffer const&), unsigned long, unsigned long, unsigned int, ucs_memory_type) (uct_p2p_test.cc:203)
==134033==    by 0x617F48: uct_p2p_test::test_xfer_multi(ucs_status_t (uct_p2p_test::*)(uct_ep*, uct_test::mapped_buffer const&, uct_test::mapped_buffer const&), unsigned long, unsigned long, unsigned int) (uct_p2p_test.cc:163)
==134033==    by 0x5F98E7: uct_p2p_rma_test_put_zcopy_Test::test_body() (test_p2p_rma.cc:111)
==134033==    by 0x543999: ucs::test_base::run() (test.cc:280)
==134033==    by 0x543B5C: ucs::test_base::TestBodyProxy() (test.cc:306)
==134033==    by 0x5DD931: uct_p2p_test::TestBody() (uct_p2p_test.h:24)
==134033==    by 0x525DDB: void testing::internal::HandleSehExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*) (gtest-all.cc:3562)
==134033==    by 0x520FB9: void testing::internal::HandleExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*) (gtest-all.cc:3598)
==134033==    by 0x5082D0: testing::Test::Run() (gtest-all.cc:3635)
==134033==    by 0x508AAB: testing::TestInfo::Run() (gtest-all.cc:3812)
==134033==    by 0x50913B: testing::TestCase::Run() (gtest-all.cc:3930)
==134033==    by 0x50F993: testing::internal::UnitTestImpl::RunAllTests() (gtest-all.cc:5808)
==134033==
1x{0..20971520} nocomp 4579
[     INFO ] memory_type:cuda-managed 0 20971520
[     SKIP ] (cuda-managed memory is not supported)
common/test.cc:265: Failure
Failed
Got 2 valgrind errors during the test
[  FAILED  ] tcp/uct_p2p_rma_test.put_zcopy/0, where GetParam() = tcp/eno1 (5875 ms)
[----------] 1 test from tcp/uct_p2p_rma_test (5887 ms total)

[----------] Global test environment tear-down
[==========] 1 test from 1 test case ran. (5962 ms total)
[  PASSED  ] 0 tests.
[  FAILED  ] 1 test, listed below:
[  FAILED  ] tcp/uct_p2p_rma_test.put_zcopy/0, where GetParam() = tcp/eno1

 1 FAILED TEST
==134033==
==134033== HEAP SUMMARY:
==134033==     in use at exit: 33,369 bytes in 73 blocks
==134033==   total heap usage: 532,887 allocs, 527,862 frees, 171,710,167 bytes allocated
==134033==
==134033== LEAK SUMMARY:
==134033==    definitely lost: 0 bytes in 0 blocks
==134033==    indirectly lost: 0 bytes in 0 blocks
==134033==      possibly lost: 0 bytes in 0 blocks
==134033==    still reachable: 33,369 bytes in 73 blocks
==134033==         suppressed: 0 bytes in 0 blocks
==134033== Reachable blocks (those to which a pointer was found) are not shown.
==134033== To see them, rerun with: --leak-check=full --show-leak-kinds=all
==134033==
==134033== For counts of detected and suppressed errors, rerun with: -v
==134033== ERROR SUMMARY: 2 errors from 1 contexts (suppressed: 288 from 2)
make: *** [test_valgrind] Error 1
make: Leaving directory `/.autodirect/mtrswgwork/alinas/workspace/tt/ucx/test/gtest'
```

Additional tests fails, can run with \*tcp\* to view all.

--- a/src/uct/tcp/tcp_ep.c
+++ b/src/uct/tcp/tcp_ep.c
@@ -831,7 +831,8 @@ static inline void uct_tcp_ep_handle_put_req(uct_tcp_ep_t *ep,
 
     ucs_assert(ep->rx.offset == ep->rx.length);
     uct_tcp_ep_ctx_rewind(&ep->rx);
-    memcpy(ep->rx.buf, put_req, sizeof(*put_req));
+    /* Since RX buffer and PUT request can be ovelapped, use memmove() */
+    memmove(ep->rx.buf, put_req, sizeof(*put_req));
     ep->ctx_caps |= UCS_BIT(UCT_TCP_EP_CTX_TYPE_PUT_RX);
 }
 

